appcues/src/main/java/com/appcues/AnalyticsPublisher.kt

```diff
@@ -0,0 +1,69 @@
+package com.appcues
+
+import com.appcues.AnalyticType.EVENT
+import com.appcues.AnalyticType.GROUP
+import com.appcues.AnalyticType.IDENTIFY
+import com.appcues.AnalyticType.SCREEN
+import com.appcues.analytics.ActivityRequestBuilder
+import com.appcues.analytics.TrackingData
+import com.appcues.data.remote.request.EventRequest
+import java.util.Date
+
+internal class AnalyticsPublisher(
+    private val storage: Storage
+) {
+
+    fun publish(listener: AnalyticsListener?, data: TrackingData) {
+        if (listener == null) return
+
+        when (data.type) {
+            EVENT -> data.request.events?.forEach {
+                listener.trackedAnalytic(EVENT, it.name, it.attributes.sanitize(), data.isInternal)
+            }
+            IDENTIFY -> listener.trackedAnalytic(IDENTIFY, storage.userId, data.request.profileUpdate, data.isInternal)
+            GROUP -> listener.trackedAnalytic(GROUP, storage.groupId, data.request.groupUpdate, data.isInternal)
+            SCREEN -> data.request.events?.forEach {
+                listener.trackedAnalytic(SCREEN, it.screenTitle(), it.attributes.sanitize(), data.isInternal)
+            }
+        }
+    }
+
+    private fun EventRequest.screenTitle(): String? =
+        attributes[ActivityRequestBuilder.SCREEN_TITLE_ATTRIBUTE] as? String
+
+    private fun Map<*, *>.sanitize(): MutableMap<String, Any> {
+        val sanitizedMap = mutableMapOf<String, Any>()
+
+        forEach {
+            val key = it.key
+            val value = it.value
+            if (key is String && value != null) {
+                sanitizedMap[key] = when (value) {
+                    // convert Date types to Double value
+                    is Date -> value.time.toDouble()
```

Comment:
Took a look at AutoPropertyDecorator `_lastContentShownAt` for example - it uses `time` (long) for the value, not an actual Date object.  So the x-plat wrappers right now also look for `Long` in the transformation to `Double`.  Maybe we need to change all usages to `Date` instead of `Long` so that this code works, or handle both `Date` and `Long` here? I don't think it would catch all the usages as currently implemented.

Reply:
I think I'm ok doing that too. so changing all Long to Double, and also removing all Date().time from the auto properties

Reply:
Ok, yeah. I think we should also update the Moshi DateAdapter we have to use Double instead of Long, for consistency.

The `Long` topic makes me think about iOS... @mmaatttt should we also add an Int64->Double translation on the iOS side so that no Int64s can sneak through to the JS wrappers? thinking custom user props for instance.

Reply:
Interesting question. I'm going with "no" because any custom props in the JS wrappers will go from JS->SDK as Doubles and not Int64s, and so it'd be doubles from SDK->JS. I can't imagine how someone'd get an Int64 into the SDK from JS.

Reply:
good point.  So @andretortolano I think that means that we don't need any Long handling here, as long as all previous usages are updated to Date, and flow through this logic.

Reply:
so nothing to do with Long. closing this one